axum/src/response/sse.rs

```diff
@@ -213,7 +213,24 @@ impl Event {
         }
 
         self.buffer.extend_from_slice(b"data: ");
-        serde_json::to_writer((&mut self.buffer).writer(), &data).map_err(axum_core::Error::new)?;
+        struct IgnoreNewLines<'a>(bytes::buf::Writer<&'a mut BytesMut>);
```

Comment:
Move this to the top of the function please. I think it's more readable to first have the helper and then the data operations, right now it's in the middle of the write operations which is a bit confusing.